runtime/lua/vim/lsp/protocol.lua

```diff
@@ -0,0 +1,936 @@
+-- Protocol for the Microsoft Language Server Protocol (mslsp)
+
+local protocol = {}
+
+local function ifnil(a, b)
+  if a == nil then return b end
+  return a
+end
+
+
+--[=[
+-- Useful for interfacing with:
```

Comment:
when is this used? to add new enums from spec revisions?

Reply:
Yeah. I basically just do `gf` on the url and then copy stuff over and use the functions to transform it.